dbt_invoke/internal/_utils.py

```diff
@@ -197,15 +194,10 @@ def dbt_ls(
         # Because we set the dbt global arg "--log-format json", if
         # line is valid json then it may be an actual result or it
         # may be some other output from dbt, like a warning.
-        try:
-            line_dict = json.loads(line)
-        # If line is not valid json, then it should be an actual
-        # result. This is because even when the "dbt ls" command
-        # arg "--output" is not set to json, non-result logs will
-        # still be in json format (due to the dbt global arg
-        # "--log-format json").
-        except ValueError:
-            result_lines_filtered.append(line)
+        data = json.loads(line).get("data")
+        if data and "msg" in data:
+            line_dict = json.loads(data["msg"])
+        else:
             continue
```

Comment:
I suggest that we keep the try/except logic since the `dbt_ls` function still accepts `output` other than json.

Here's one way I think we can combine it with your new logic:
```suggestion
        try:
            line_dict = json.loads(line)
        # If line is not valid json, then it should be an actual
        # result. This is because even when the "dbt ls" command
        # arg "--output" is not set to json, non-result logs will
        # still be in json format (due to the dbt global arg
        # "--log-format json").
        except ValueError:
            result_lines_filtered.append(line)
            continue
        data = line_dict.get("data")
        if data and "msg" in data:
            line_dict = json.loads(data["msg"])
        else:
            continue
```

Reply:
agree, makes the function more universal